src/vs/editor/common/commands/shiftCommand.ts

```diff
@@ -99,7 +105,23 @@ export class ShiftCommand implements EditorCommon.ICommand {
 				indents[j] = indents[j-1] + oneIndent;
 			}
 
-			builder.addEditOperation(new Range(lineNumber, 1, lineNumber, indentationEndIndex + 1), indents[desiredIndentCount]);
+			var desiredIndent = indents[desiredIndentCount];
+			if (isMultilineComment) {
+				// Add space to align the multiline comment
+				desiredIndent += " ";
+			}
+
+			// Check whether we're in a multiline comment or not
+			if (lineText.length >= 2 + indentationEndIndex) {
+				if (lineText.indexOf("/*") === indentationEndIndex) {
+					isMultilineComment = true;
+				}
+				if (lineText.indexOf("*/") !== -1) {
```

Comment:
[minor] You can pass the start index in the call to indexOf. Will reduce atleast 2 comparisons.